core/dbt/include/global_project/macros/materializations/models/table/columns_spec_ddl.sql

```diff
@@ -9,13 +9,24 @@
 {% macro columns_spec_ddl() %}
   {# loop through user_provided_columns to create DDL with data types and constraints #}
     {%- set user_provided_columns = model['columns'] -%}
+    {%- set unsupported_contraints = adapter.list_unsupported_constraints() -%}
+    {%- set all_constraints = [] -%}
     (
-    {% for i in user_provided_columns %}
+    {% for i in user_provided_columns -%}
       {%- set col = user_provided_columns[i] -%}
       {%- set constraints = col['constraints'] -%}
-      {{ col['name'] }} {{ col['data_type'] }}{% for c in constraints %} {{ adapter.render_raw_column_constraint(c) }}{% endfor %}{{ "," if not loop.last }}
-    {% endfor -%}
+      {{ col['name'] }} {{ col['data_type'] }}
+      {% for c in constraints -%}
+        {% do all_constraints.append(c) %}
+        {%- if c["type"] not in unsupported_contraints -%}
```

Comment:
I'm thinking we can move a lot of this filtering logic to the python base implementation, potentially within the existing `adapter.render_raw_column_constraint` method. Motivation being - this filtering / validation logic should be consistent across adapters (with flexibility in configuring _which_ constraints are supported/enforced). There's only one correct way to render this DDL for a given adapter - so any overrides should ideally be implemented by the adapter-maintainer in python, as opposed to by end users in "user-space" (jinja).